public/app/core/components/NestedFolderPicker/NestedFolderPicker.test.tsx

```diff
@@ -174,14 +167,36 @@ describe('NestedFolderPicker', () => {
   });
 
   it('hides folders specififed by UID', async () => {
-    render(<NestedFolderPicker excludeUIDs={[folderA.item.uid]} onChange={mockOnChange} />);
+    render(<NestedFolderPicker excludeUIDs={[folderC.item.uid]} onChange={mockOnChange} />);
 
     // Open the picker and wait for children to load
     const button = await screen.findByRole('button', { name: 'Select folder' });
     await userEvent.click(button);
-    await screen.findByLabelText(folderB.item.title);
+    await screen.findByLabelText(folderA.item.title);
 
-    expect(screen.queryByLabelText(folderA.item.title)).not.toBeInTheDocument();
+    expect(screen.queryByLabelText(folderC.item.title)).not.toBeInTheDocument();
+  });
+
+  it('by default only shows items the user can edit', async () => {
+    render(<NestedFolderPicker onChange={mockOnChange} />);
+
+    const button = await screen.findByRole('button', { name: 'Select folder' });
+    await userEvent.click(button);
+    await screen.findByLabelText(folderA.item.title);
+
+    expect(screen.queryByLabelText(folderB.item.title)).not.toBeInTheDocument(); // folderB is not editable
+    expect(screen.queryByLabelText(folderC.item.title)).toBeInTheDocument(); // but folderC is
+  });
+
+  it('shows items the user can view, with the prop', async () => {
+    render(<NestedFolderPicker permission={PermissionLevelString.View} onChange={mockOnChange} />);
+
+    const button = await screen.findByRole('button', { name: 'Select folder' });
+    await userEvent.click(button);
+    await screen.findByLabelText(folderA.item.title);
+
+    expect(screen.queryByLabelText(folderB.item.title)).toBeInTheDocument();
+    expect(screen.queryByLabelText(folderC.item.title)).toBeInTheDocument();
```

Comment:
same comments here re: wrapping in expect and using `getBy` instead of `queryBy`